SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -244,12 +247,78 @@ TArray<UClass*> GetAllSupportedClasses()
 	return Classes.Array();
 }
 
-bool SpatialGDKGenerateSchema()
+void DeleteGeneratedSchemaFiles()
+{
+	const FString SchemaOutputPath = GetDefault<USpatialGDKEditorSettings>()->GetGeneratedSchemaOutputFolder();
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	if (PlatformFile.DirectoryExists(*SchemaOutputPath))
+	{
+		if (!PlatformFile.DeleteDirectoryRecursively(*SchemaOutputPath))
+		{
+			UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("Could not clean the generated schema directory '%s'! Please make sure the directory and the files inside are writeable."), *SchemaOutputPath);
+		}
+	}
+	PlatformFile.CreateDirectory(*SchemaOutputPath);
+}
+
+void InitClassPathToSchemaMap()
 {
-	ClassPathToSchema.Empty();
+	TSoftObjectPtr<USchemaDatabase> SchemaDatabasePtr(FSoftObjectPath(TEXT("/Game/Spatial/SchemaDatabase.SchemaDatabase")));
+	SchemaDatabasePtr.LoadSynchronous();
+	const USchemaDatabase* const SchemaDatabase = SchemaDatabasePtr.Get();
 
-	SchemaGeneratedClasses = GetAllSupportedClasses();
+	if (SchemaDatabase)
+	{
+		ClassPathToSchema = SchemaDatabase->ClassPathToSchema;
+		NextAvailableComponentId = SchemaDatabase->NextAvailableComponentId;
+	}
+	else
+	{
+		NextAvailableComponentId = SpatialConstants::STARTING_GENERATED_COMPONENT_ID;
+		// As a safety precaution, if the SchemaDatabase.uasset doesn't exist then make sure the schema generated folder is cleared as well. 
```

Comment:
Can you log here that no SchemaDatabase is was found so we will be clearing out the old folder (if it exists) and generating a new one?